src/components/CalendarMonthGrid.jsx

```diff
@@ -73,6 +76,23 @@ export default class CalendarMonthGrid extends React.Component {
     this.container.addEventListener('transitionend', this.onTransitionEnd);
   }
 
+  componentWillReceiveProps(nextProps) {
+    const { initialMonth, numberOfMonths } = nextProps;
+    const { months } = this.state;
+
+    const hasMonthChanged = !this.props.initialMonth.isSame(initialMonth, 'month');
+    const hasNumberOfMonthsChanged = this.props.numberOfMonths !== numberOfMonths;
+    if (hasMonthChanged || hasNumberOfMonthsChanged) {
+      if (initialMonth.isAfter(this.props.initialMonth)) {
+        months.shift();
```

Comment:
This definitely needs to be changed. You should use the callback version of setState:

```js
this.setState((previousState) => {
  ...
  return {...};
});
```